### PR TITLE
Ben

### PR DESCRIPTION
**Pull request by bbjohnson6810**
multiple new utility functions

### CODE REVIEW

**Review by robert-vogel — DISMISSED** [review]
Looks good.

File: R/utils.R [hunk 1]
```diff
@@ -205,3 +205,63 @@ argument_parser <- function(..., description=NULL)
...
+#'      phenotype, if only processing one. If NULL, all non-ID
+#'      columns in the file will be processed. 
+#' @param output_dir (string)
+#'      The directory in which the Plink-formatted IDs file will be 
+#'      saved.
+#'
+#' @return A list containing (1) the file path to the Plink-formatted 
+#'      IDs file and (2) a vector of phenotyped IDs.
+#
+get_phenotyped_ids <- function(phenotype_data, id_column, trait_column = NULL, output_dir) {
```
Comment (robert-vogel) [inline R/utils.R:238]: I think that named arguments with default values go after all required arguments.  Consider
`
get_phenotyped_ids <- function(phenotype_data, id_column, output_dir, trait_column = NULL) {`

File: R/utils.R [hunk 2]
```diff
@@ -205,3 +205,63 @@ argument_parser <- function(..., description=NULL)
...
+    
+        write.table(data.frame(fam = 0, id = pheno_dat[[id_column]]), 
+                phtyped_ids_file, sep='\t', row.names=F, col.names=F, quote=F)
+
+        return(list(ids_file = phtyped_ids_file, ids = as.character(pheno_dat[[id_column]])))
+
+    } else {
+    
+        trait <- trait_column
+        phtyped_ids_file <- paste0(output_dir, '/phtyped_ids_', trait)
```
Comment (robert-vogel) [inline R/utils.R:254]: Change paste0 to file.path for constructing paths.

File: R/utils.R [hunk 3]
```diff
@@ -205,3 +205,63 @@ argument_parser <- function(..., description=NULL)
...
+#' @return A list containing (1) the file path to the Plink-formatted 
+#'      IDs file and (2) a vector of phenotyped IDs.
+#
+get_phenotyped_ids <- function(phenotype_data, id_column, trait_column = NULL, output_dir) {
+
+    pheno_dat <- read.csv(phenotype_data)
+
+    if (is.null(trait_column)) {
+    
+        phtyped_ids_file <- paste0(output_dir, '/phtyped_ids_all')
```
Comment (robert-vogel) [inline R/utils.R:244]: Change paste0 to file.path for constructing paths.

**Review by robert-vogel — CHANGES_REQUESTED** [review]
Take a look at some of the requested changes.  This is to be an exercise for the review process, I won't be this picky in future reviews.

File: R/utils.R (hunk 1 above)
Comment (bbjohnson6810) [reply R/utils.R]: addressed all comments in [d2cc608](https://github.com/Palmer-Lab-UCSD/rattaca/pull/7/commits/d2cc608948b7bad2f5fa8c1bb2b5498b4d1a8e0e)

Comment (robert-vogel) [reply R/utils.R]: Looks good!

**Review by robert-vogel — COMMENTED** [review]
Overall the code in commit https://github.com/Palmer-Lab-UCSD/rattaca/pull/7/commits/82cdd4921f2bc205329e2229f02850bd54ea80dc is clear and well planned.  I have some minor comments that you are free to reject, accept, or transform.

**Pull request merged**